.github/workflows/e2e.yml

```diff
@@ -0,0 +1,298 @@
+      - name: Download artifact
+        uses: actions/download-artifact@v3
+        with:
+          names: api-server coordinator dashboard dashboard-metadata metadata serving worker
+          path: /tmp
+
+      - name: Check Artifacts
+        run: |
+          ls /tmp
+          ls /tmp/api-server
+
+      - name: Install grpc_tools
+        run: pip install grpcio-tools
+
+      - name: Install Protobuf
+        run: sudo snap install protobuf --classic
+
+      - name: Setup Proto
+        run: ./gen_grpc.sh
+
+      - name: Install python dependencies
+        working-directory: ./
+        run: |
+          python3 -m pip install build pytest
+          python3 -m build ./client/
+          python3 -m pip install client/dist/*
+
+      - name: Start minikube
+        run: |
+          curl -LO \
+            https://storage.googleapis.com/minikube/releases/latest/minikube-linux-amd64 \
+            && install minikube-linux-amd64 /tmp/
+  
+            kv=$(curl -s https://storage.googleapis.com/kubernetes-release/release/stable.txt)
+            curl -LO \
+            https://storage.googleapis.com/kubernetes-release/release/$kv/bin/linux/amd64/kubectl \
+            && install kubectl /tmp/
+  
+            /tmp/minikube-linux-amd64 config set WantUpdateNotification false
+            /tmp/minikube-linux-amd64 start --driver=docker
+
+
+      - name: Configure Docker Socket
+        run: eval $(minikube docker-env)
+
+      - name: Load image
+        run: |
+          docker load --input /tmp/api-server/api-server.tar
+          docker load --input /tmp/coordinator/coordinator.tar
+          docker load --input /tmp/dashboard/dashboard.tar
+          docker load --input /tmp/metadata-dashboard/metadata-dashboard.tar
+          docker load --input /tmp/metadata/metadata.tar
+          docker load --input /tmp/serving/serving.tar
+          docker load --input /tmp/worker/worker.tar
+          minikube image load local/api-server:stable
+          minikube image load local/coordinator:stable
+          minikube image load local/dashboard:stable
+          minikube image load local/metadata-dashboard:stable
+          minikube image load local/metadata:stable
+          minikube image load local/serving:stable
+          minikube image load local/worker:stable
+          minikube image ls
+
+      - name: Helm Install
+        run: |
+          helm repo add jetstack https://charts.jetstack.io
+          helm repo update
+          helm install certmgr jetstack/cert-manager --set installCRDs=true --version v1.8.0 --namespace cert-manager --create-namespace
+          helm install featureform ./charts/featureform --set global.repo=local --set global.pullPolicy=Never
+          helm install quickstart ./charts/quickstart
+
+      - name: Check Pods
+        run: |
+          seconds=0
+          while [ $(kubectl get pods | grep "featureform" | grep -wv featureform-quickstart | awk {'print $3'} | column -t | grep -wv Running -c ) != 0 ]
```

Comment:
you can query pods based on a label set in the config, or based on their service name if specified in the charts. Set by the metadata labels in the yaml files

kubectl get pods -l app=featureform

avoids needing to pattern match indirectly

Also kubectl has a wait command which pauses until a queried container is ready

kubectl wait --for-condition=ready pod -l app=featureform

Since this works though it's fine

Reply:
Changed this